pwiz_tools/Skyline/Executables/SkylineBatch/SkylineBatch/ConfigRunner.cs

```diff
@@ -36,15 +44,17 @@ public class ConfigRunner : IConfigRunner
         private readonly object _lock = new object();
         private RunnerStatus _runnerStatus;
         private readonly ProcessRunner _processRunner;
-        private List<string> _batchCommandsToLog;
+        private string _batchFile;
+
+        private List<CancellationTokenSource> _runningCancellationTokens;

```

Comment:
You only need one CancellationTokenSource, since you never want to cancel a subset of the things that are happening.